pyrf/gui/gui.py

```diff
@@ -454,8 +503,15 @@ def _add_docking_controls(self, widget, title):
         dock.setWidget(widget)
         # FIXME we should be doing this in the MainWindow
         self._main_window.addDockWidget(QtCore.Qt.RightDockWidgetArea, dock)
-        self._main_window.view_menu.addAction(dock.toggleViewAction())
-
+        toggle_action = dock.toggleViewAction()
+        def _dock_change():
+            dict = {title.lower().replace(' ', '_'): toggle_action.isChecked()}
```

Comment:
don't shadow `dict` builtin (use a variable different name)

Also this `title.lower().replace(' ', '_')` is repeated a few times, maybe make a tiny function with a clear name `title_to_options_name(title)` or similar